### PR TITLE
Teleop control refactor

### PR DESCRIPTION
**Pull request by witcpalek**
1. Make all set methods private
2. Initialize the buttons/axes using a map which contains the scaling, deadband, inversion, etc.
3. Delay initializing the controllers until teleopinit
4. Deal with round limit on the joysticks (if the joystick is at 45 degrees and maxed out, previously it would return .707 for both axes.  Now it will return 1.0.
5. Break out teleop enums, structures, functions into their own files
6. Move Logger and TeleopControl to their own folders

### CODE REVIEW

File: src/main/cpp/gamepad/DragonXBox.h [hunk 1]
```diff
@@ -17,11 +17,17 @@
...
+
 // FRC includes
 #include <frc/XboxController.h>
 
 // Team 302 includes
 #include <gamepad/IDragonGamePad.h>
+#include <teleopcontrol/TeleopControlMappingEnums.h>
+
+
+#include <RobinHood/robin_hood.h>
```
Comment (nickfrisch) [inline src/main/cpp/gamepad/DragonXBox.h:30]: The robin hood map defined below looks like it was commented out and is no longer in use, is this include still needed?

File: src/main/cpp/gamepad/DragonXBox.cpp [hunk 2]
```diff
@@ -28,7 +30,10 @@
 #include <gamepad/button/POVButton.h>
 #include <gamepad/button/ToggleButton.h>
 #include <gamepad/DragonXBox.h>
-#include <utils/Logger.h>
+#include <utils/logging/Logger.h>
+#include <teleopcontrol/TeleopControlMappingEnums.h>
+
+#include<RobinHood/robin_hood.h>
```
Comment (nickfrisch) [inline src/main/cpp/gamepad/DragonXBox.cpp:36]: Is this include still needed?

File: src/main/cpp/gamepad/DragonXBox.h (hunk 1 above)
Comment (witcpalek) [reply src/main/cpp/gamepad/DragonXBox.h:30]: removed

File: src/main/cpp/gamepad/DragonXBox.cpp (hunk 2 above)
Comment (witcpalek) [reply src/main/cpp/gamepad/DragonXBox.cpp:36]: removed

**Pull request merged**